components/xip-patch/xip-patch.sh

```diff
@@ -2,77 +2,112 @@
 
 set -o errexit
 
-if [ -z "${INGRESSGATEWAY_SERVICE_NAME}" ]; then
-  INGRESSGATEWAY_SERVICE_NAME=istio-ingressgateway
-fi
-
-if [ -z "${EXTERNAL_PUBLIC_IP}" ]; then
-
-  SECONDS=0
-  END_TIME=$((SECONDS+60))
+# # # # # # # # # # # # # # # # #
+# VARs coming from environment: #
+#                               #
+# EXTERNAL_PUBLIC_IP            #
+# INGRESSGATEWAY_SERVICE_NAME   #
+# PUBLIC_DOMAIN                 #
+# TLS_CERT                      #
+# TLS_KEY                       #
+# # # # # # # # # # # # # # # # #
 
-  while [ ${SECONDS} -lt ${END_TIME} ];do
-      echo "Trying to get loadbalancer IP address"
+function generateXipDomain() {
```

Comment:
Function must take explicit input parameters instead of "global" env names, it has to be reusable.
Explanation: We'll need xip "patch" functionality for at least two cases now: istio-ingress and application-connector-ingress. This function should handle both cases.